fxa/plugins/requests.py

```diff
@@ -92,18 +125,29 @@ def __init__(self, email, password, scopes=None, client_id=None,
         self.account_server_url = account_server_url
         self.oauth_server_url = oauth_server_url
 
-    def __call__(self, request):
-        client = core.Client(server_url=self.account_server_url)
-        session = client.login(self.email, self.password, keys=True)
-
-        url = urlparse(self.oauth_server_url)
-        audience = "%s://%s/" % (url.scheme, url.netloc)
+        self.cache = cache
```

Comment:
Wouldn't it be better to use cache as an object if present? (if one would like to use a different cache than the memory one).